develop/blockchains/maintenance/storage-migrations.md

```diff
@@ -1,4 +1,183 @@
 ---
 title: Storage Migrations
-description: TODO
----
\ No newline at end of file
+description: Ensure smooth runtime upgrades with storage migrations, updating data formats and preventing errors. Learn when and how to implement migrations efficiently.
+---
+
+# Storage Migrations
+
+## Introduction
+
+Storage migrations are a crucial part of the runtime upgrade process. They allow you to update the storage layout of your blockchain, adapting to changes in the runtime. Whenever you change the encoding or data types used to represent data in storage, you'll need to provide a storage migration to ensure the runtime can properly interpret the existing stored values in the new runtime state.
+
+Storage migrations must be executed at a precise moment during the runtime upgrade process to ensure data consistency and prevent runtime panics. The migration code needs to run:
+
+- After the new runtime is deployed
+- Before any other code from the new runtime executes
+- Before any [`on_initialize`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.Hooks.html#method.on_initialize){target=\_blank} hooks run
+- Before any transactions are processed
+
+This timing is critical because the new runtime expects data to be in the updated format. Without proper migration, any attempt to decode the old data format could result in runtime panics or undefined behavior.
+
+## When is a Storage Migration Required?
+
+A storage migration is necessary whenever a runtime upgrade changes the storage layout or the encoding/interpretation of existing data. Even if the underlying data type appears to still "fit" the new storage representation, a migration may be required if the interpretation of the stored values has changed.
+
+- Adding or removing an extrinsic introduces no new interpretation of preexisting data, so not migration required
+
+- Reordering or mutating fields of an existing data type do change the encoded/decoded data representation, hence a storage migration is needed
+
+- Removal of a pallet or storage item warrants cleaning up storage via a migration to avoid state bloat
+
+- Adding a new storage item would not require any migration
+
+### Common Cases
+
+Here are some common scenarios where a storage migration is needed:
+
+- Changing data types:
+
+    ```rust
+    --8<-- 'code/develop/blockchains/maintenance/storage-migrations/example-1.rs'
+    ```
+
+    Changing the underlying data type requires a migration to convert the existing values.
```

Comment:
I'd dive deeper into the example to make it even more straightforward...

Thinking out loud:
"In this case, the example showcases a change from u32 data type to u64, however, this can happen with any allowed rust type."